src/Compilers/CSharp/Portable/Binder/Binder_Constraints.cs

```diff
@@ -86,34 +86,64 @@ internal partial class Binder
         /// <summary>
         /// Bind and return a single type parameter constraint clause.
         /// </summary>
-        private TypeParameterConstraintClause BindTypeParameterConstraints(string name, SeparatedSyntaxList<TypeParameterConstraintSyntax> constraintsSyntax, DiagnosticBag diagnostics)
+        private TypeParameterConstraintClause BindTypeParameterConstraints(
+            string name, SeparatedSyntaxList<TypeParameterConstraintSyntax> constraintsSyntax, DiagnosticBag diagnostics)
         {
             var constraints = TypeParameterConstraintKind.None;
             var constraintTypes = ArrayBuilder<TypeSymbol>.GetInstance();
+            var isStruct = false;
 
-            foreach (var syntax in constraintsSyntax)
+            for (int i = 0, n = constraintsSyntax.Count; i < n; i++)
             {
+                var syntax = constraintsSyntax[i];
                 switch (syntax.Kind())
                 {
                     case SyntaxKind.ClassConstraint:
+                        if (i != 0)
+                        {
+                            diagnostics.Add(ErrorCode.ERR_RefValBoundMustBeFirst, syntax.GetFirstToken().GetLocation());
+                        }
+
                         constraints |= TypeParameterConstraintKind.ReferenceType;
-                        break;
```

Comment:
>break; [](start = 24, length = 6)

Is there a special reason to use ```continue``` here? I think it is better to use ```break``` and rely on compiler to optimize.